pkg/controller/pod_test.go

```diff
@@ -244,6 +246,61 @@ var _ = Describe("Dynamic Attachment controller", func() {
 					Eventually(<-eventRecorder.Events).Should(Equal(expectedEventPayload))
 				})
 			})
+
+			When("an attachment is added with attributes (IPs, MAC, cni-args)", func() {
+				JustBeforeEach(func() {
+					pod = updatePodSpec(pod)
+					netSelectionElements := append(generateNetworkSelectionElements(namespace, networkName),
+						nad.NetworkSelectionElement{
+							Name:             networkToAdd,
+							Namespace:        namespace,
+							InterfaceRequest: "net1",
+							IPRequest:        []string{ipAddr},
+							MacRequest:       macAddr,
+							CNIArgs: &map[string]interface{}{
+								"foo": "bar",
+							},
+						},
+					)
+					serelizedNetSelectionElements, _ := json.Marshal(netSelectionElements)
+					pod.Annotations[nad.NetworkAttachmentAnnot] = string(serelizedNetSelectionElements)
+					_, err := k8sClient.CoreV1().Pods(namespace).UpdateStatus(
+						context.TODO(),
+						pod,
+						metav1.UpdateOptions{})
+					Expect(err).NotTo(HaveOccurred())
+				})
+
+				It("an `AddedInterface` event is seen in the event recorded", func() {
+					expectedEventPayload := fmt.Sprintf(
+						"Normal AddedInterface pod [%s]: added interface %s to network: %s (ips: [%s], mac: %s, cni-args: %v)",
+						annotations.NamespacedName(namespace, podName),
+						"net1",
+						networkToAdd,
+						ipAddr,
+						macAddr,
+						cniArgs,
+					)
+					Eventually(<-eventRecorder.Events).Should(Equal(expectedEventPayload))
+				})
+
+				It("the pod network-status is updated with the new network attachment", func() {
+					Eventually(func() ([]nad.NetworkStatus, error) {
+						updatedPod, err := k8sClient.CoreV1().Pods(namespace).Get(context.TODO(), podName, metav1.GetOptions{})
+						if err != nil {
+							return nil, err
+						}
+						status, err := annotations.PodDynamicNetworkStatus(updatedPod)
+						if err != nil {
+							return nil, err
+						}
+						return status, nil
+					}).Should(ConsistOf(
+						ifaceStatus(namespace, networkName, "net0", ""),
+						ifaceStatus(namespace, networkToAdd, "net1", macAddr)))
```

Comment:
given the linter's complaints, I would just define a const within the ifaceStatus func, and rename it to `ifaceStatusForDefaultNamespace`. 

... not the most elegant solution but it would unlock your PR. 

I would refactor the tests afterwards to address that.